scripts/nn/test/test.dml

```diff
@@ -410,6 +411,64 @@ cross_entropy_loss = function() {
   }
 }
 
+cross_entropy_loss2d = function() {
+  /*
+   * Test for the 2D cross-entropy loss function.
+   */
+  print("Testing the forward part of the 2D cross-entropy loss function")
+
+  # Generate data
+  N = 2  # num examples
+  C = 3  # num targets
+  Hin = 3  # example height
+  Win = 3  # example width
+  loss_expected = 0.0770996
+
+  # pred data after the softmax
+  pred = matrix("9.99909163e-01   4.99988675e-01   4.53958055e-05
+                 9.99909163e-01   4.53958055e-05   4.53958055e-05
+                 9.99909163e-01   4.53958055e-05   4.53958055e-05
+                 4.53958055e-05   4.99988675e-01   4.53958055e-05
+                 4.53958055e-05   9.99909163e-01   4.53958055e-05
+                 4.53958055e-05   9.99909163e-01   4.53958055e-05
+                 4.53958055e-05   2.26994507e-05   9.99909163e-01
+                 4.53958055e-05   4.53958055e-05   9.99909163e-01
+                 4.53958055e-05   4.53958055e-05   9.99909163e-01
+                 9.99909163e-01   4.99988675e-01   4.53958055e-05
+                 9.99909163e-01   4.53958055e-05   4.53958055e-05
+                 9.99909163e-01   4.53958055e-05   4.53958055e-05
+                 4.53958055e-05   4.99988675e-01   4.53958055e-05
+                 4.53958055e-05   9.99909163e-01   4.53958055e-05
+                 4.53958055e-05   9.99909163e-01   4.53958055e-05
+                 4.53958055e-05   2.26994507e-05   9.99909163e-01
+                 4.53958055e-05   4.53958055e-05   9.99909163e-01
+                 4.53958055e-05   4.53958055e-05   9.99909163e-01", rows=N, cols=C*Hin*Win)
+  y = matrix("1 0 0
+              1 0 0
+              1 0 0
+              0 1 0
+              0 1 0
+              0 1 0
+              0 0 1
+              0 0 1
+              0 0 1
+              1 0 0
+              1 0 0
+              1 0 0
+              0 1 0
+              0 1 0
+              0 1 0
+              0 0 1
+              0 0 1
+              0 0 1", rows=N, cols=C*Hin*Win)
+
+  loss = cross_entropy_loss2d::forward(pred, y, C)
+
+  if (abs(loss - loss_expected) > 1.0e-7) {
+    print("ERROR: The 2D cross-entropy loss function ouptut (" + loss + ") does not match the expected value " + loss_expected + ".")
+  }
```

Comment:
For testing consistency, can we use the `test_util::check_rel_error` function instead here, probably with `1e-3` & `1e-4` for the error and warning threshold arguments like the other tests?